src/java/org/apache/cassandra/cql3/restrictions/RestrictionSetWrapper.java

```diff
@@ -54,25 +52,26 @@ public void addToRowFilter(RowFilter filter,
         restrictions.addToRowFilter(filter, indexRegistry, options);
     }
 
-    public List<ColumnMetadata> getColumnDefs()
+    public List<ColumnMetadata> columns()
     {
-        return restrictions.getColumnDefs();
+        return restrictions.columns();
     }
 
-    @Override
-    public Collection<ColumnMetadata> getColumnDefinitions()
```

Comment:
I know `getColumnDefinitions()` looks like a duplicate, but I think it was added in CASSANDRA-18155 explicitly to avoid defensive copying the `keySet()` in `RestrictionSet`.

Reply:
Maybe we should add a comment so that the next patch doesn't delete it too :-)

Reply:
I will take the blame for this :D

Reply:
+1000 to commenting it

Reply:
The method is not needed any more. The patch changed the logic. Instead of iterating over the columns it uses the internal maps to locate the restrictions an check their type. It should be more efficient than the previous approach.
It seems that this information could be precomputed at preparation time and stored. I will keep it in mind for the next round of refactoring.